framework/db/src/com/cloud/utils/db/TransactionLegacy.java

```diff
@@ -1014,6 +1014,13 @@ public static void initDataSource(Properties dbProps) {
             if (dbProps.size() == 0)
                 return;
 
+            try {
+                // Explicitely load JDBC driver because it has been removed from the classpath as part of commit c22659d76d73f00f41c13776c490e17a50aacd20
+                Class.forName("com.mysql.jdbc.Driver");
```

Comment:
1. You are absolutely right, but I explicitly loaded this one because "mysql:jdbc" is hardcoded in the connection URL as well.
2. The problem is that the JAR is actually loaded (so Class.forName works), but the JDBC driver is not auto-registering itself. Tomcat has a really specific way of auto-registering its JDBC drivers.. https://tomcat.apache.org/tomcat-7.0-doc/jndi-datasource-examples-howto.html#DriverManager,_the_service_provider_mechanism_and_memory_leaks